tests_metricflow/dataflow/optimizer/test_predicate_pushdown_optimizer.py

```diff
@@ -148,3 +160,238 @@ def test_applied_filter_back_propagation(branch_state_tracker: PredicatePushdown
+    """
+    query_spec = query_parser.parse_and_validate_query(
+        metric_names=("visit_buy_conversion_rate_7days",),
+        group_by_names=("metric_time", "user__home_state_latest"),
+        where_constraint=PydanticWhereFilter(where_sql_template="{{ Dimension('visit__referrer_id') }} = '123456'"),
+    ).query_spec
+    _check_optimization(
+        request=request,
+        mf_test_configuration=mf_test_configuration,
+        dataflow_plan_builder=dataflow_plan_builder,
+        query_spec=query_spec,
+        expected_additional_constraint_nodes_in_optimized=1,  # TODO: Remove superfluous where constraint nodes
+    )
+
+
+def test_cumulative_metric_predicate_pushdown(
+    request: FixtureRequest,
+    mf_test_configuration: MetricFlowTestConfiguration,
+    query_parser: MetricFlowQueryParser,
+    dataflow_plan_builder: DataflowPlanBuilder,
+) -> None:
+    """Tests pushdown optimizer behavior for a query against a cumulative metric.
+
+    At this time categorical dimension predicates should be pushed down, but metric_time predicates should not be,
+    since supporting time filter pushdown for cumulative metrics requires filter expansion to ensure we capture the
+    full set of inputs for the initial cumulative window.
+
+    TODO: Add metric time filters
+    """
+    query_spec = query_parser.parse_and_validate_query(
+        metric_names=("every_two_days_bookers",),
+        group_by_names=("listing__country_latest", "metric_time"),
+        where_constraint=PydanticWhereFilter(where_sql_template="{{ Dimension('booking__is_instant') }}"),
+    ).query_spec
+    _check_optimization(
+        request=request,
+        mf_test_configuration=mf_test_configuration,
+        dataflow_plan_builder=dataflow_plan_builder,
+        query_spec=query_spec,
+        expected_additional_constraint_nodes_in_optimized=1,
+    )
+
+
+@pytest.mark.skip("plan output has non-deterministic ordering")
+def test_aggregate_output_join_metric_predicate_pushdown(
+    request: FixtureRequest,
+    mf_test_configuration: MetricFlowTestConfiguration,
+    query_parser: MetricFlowQueryParser,
+    dataflow_plan_builder: DataflowPlanBuilder,
+) -> None:
+    """Tests pushdown optimizer behavior when a metric does an aggregate output metric join.
+
+    In this case we expect filters to not be pushed down, since they are outside of a full outer join.
+    """
+    query_spec = query_parser.parse_and_validate_query(
+        metric_names=("views_times_booking_value",),
+        where_constraint=PydanticWhereFilter(where_sql_template="{{ Dimension('listing__is_lux_latest') }}"),
+    ).query_spec
+    _check_optimization(
+        request=request,
+        mf_test_configuration=mf_test_configuration,
+        dataflow_plan_builder=dataflow_plan_builder,
+        query_spec=query_spec,
+        expected_additional_constraint_nodes_in_optimized=0,
+    )
+
+
+def test_offset_metric_predicate_pushdown(
```

Comment:
I think we also have an extra where filter applied to cumulative metrics with `grain_to_date`, so might be good to test that specifically!

Reply:
As far as I can tell those are only for time constraints, not general where constraints, but I'll add a test case to the top of the stack.